src/sentry/integrations/gitlab/repository.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.plugins import providers
+from sentry.models import Integration
+
+
+class GitlabRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Gitlab'
+
+    def get_installation(self, integration_id, organization_id):
+        if integration_id is None:
+            raise ValueError('%s requires an integration_id' % self.name)
+
+        try:
+            integration_model = Integration.objects.get(id=integration_id)
+        except Integration.DoesNotExist as error:
+            self.handle_api_error(error)
+
+        return integration_model.get_installation(organization_id)
+
+    def validate_config(self, organization, config):
```

Comment:
This function isn't really validating anything – would be nice to describe it's purpose a little more.

Reply:
It should be to validate the data that comes in. This is really a carry-over from the plugin functionality. Github, I think, does in fact validate something. I don't really validate anything here unless you agree that making an api call to get the repo is validating the information. I could add the code potentially to the create repo method.

Reply:
If this is like, a subclass method that _neesd_ to be named `validate_config` because that's just the way this all works, then I'd write a quick comment explaining that.

If not, then maybe renaming it to `extract_config` is more appropriate.